### PR TITLE
TASK-51210 :  First screen of the App

### PR DESCRIPTION
**Pull request by GouadriaHanen**

### CODE REVIEW

File: challenges-api/src/main/java/org/exoplatform/challenges/services/ChallengeService.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import org.exoplatform.services.security.Identity;
+
+public interface ChallengeService {
+    
+    /**
+     * Return a boolean that indicates if the current user can add a challegne or not
+     *
+     * @param currentUser
+     * @return if the user can add a challenge or not    */
+    boolean canAddChallenge( String currentUser) throws Exception;
```
Comment (ahamdi) [inline challenges-api/src/main/java/org/exoplatform/challenges/services/ChallengeService.java]:
```suggestion
    boolean canAddChallenge(String currentUser) throws Exception;
```

File: challenges-services/src/main/java/org/exoplatform/challenges/services/ChallengeServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import org.exoplatform.services.organization.OrganizationService;
+import org.exoplatform.social.core.space.spi.SpaceService;
+
+public class ChallengeServiceImpl implements ChallengeService {
+
+
+    private  OrganizationService organizationService ;
+
+    public ChallengeServiceImpl(OrganizationService organizationService) {
+        this.organizationService = organizationService;
```
Comment (ahamdi) [inline challenges-services/src/main/java/org/exoplatform/challenges/services/ChallengeServiceImpl.java]: Do we really need the organizationService ?

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: challenges-webapp/src/main/webapp/vue-app/challenges/challengesServices.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export function canAddChallenge() {
+  return fetch(`${eXo.env.portal.context}/${eXo.env.portal.rest}/challenge/api/canAddChallenge`, {
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    method: 'GET'
+  }).then((resp) => resp.json()).then(resp => {
+    return resp;
```
Comment (Rdidmohamedelhadi) [inline challenges-webapp/src/main/webapp/vue-app/challenges/challengesServices.js]:
```suggestion
          if (resp && resp.ok) {
        return resp.json();
      } else {
        throw new Error('Server indicates an error while sending request');
      }
```

File: challenges-webapp/src/main/webapp/vue-app/challenges/components/AddChallenge.vue [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<template>
+  <div class="border-box-sizing clickable addChallengeButton">
+    <v-btn class="btn btn-primary">
+      <i class="fas fa-plus pr-1"></i>
+      <span class="ms-2 d-none d-lg-inline">
+        {{ $t('exoplatform.challenges.button.addChallenge') }}
```
Comment (Rdidmohamedelhadi) [inline challenges-webapp/src/main/webapp/vue-app/challenges/components/AddChallenge.vue]: It's better to always use simple keys for labels:
exoplatform.challenges.button.addChallenge = > challenges.button.addChallenge

File: challenges-webapp/src/main/webapp/vue-app/challenges/components/WelcomeMessage.vue [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<template>
+  <div class="welcomeMessage">
+    <img
+      src="/challenges/images/Challenge_app_backg_image.png"
+      class="mx-4"
+      loading="lazy">
```
Comment (Rdidmohamedelhadi) [inline challenges-webapp/src/main/webapp/vue-app/challenges/components/WelcomeMessage.vue]: please add alt

File: challenges-webapp/src/main/webapp/vue-app/challenges/components/WelcomeMessage.vue [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    <div class="mx-4 my-6">
+      <p class="title"> {{ $t('exoplatform.challenges.message.welcomeMessage') }} </p>
+      <p v-if="canAddChallenge"> {{ $t('exoplatform.challenges.message.welcomeMessageForManager') }} </p>
+      <p v-else> {{ $t('exoplatform.challenges.message.welcomeMessageForUser') }} </p>
```
Comment (Rdidmohamedelhadi) [inline challenges-webapp/src/main/webapp/vue-app/challenges/components/WelcomeMessage.vue]: exoplatform.challenges.message.welcomeMessage ==>challenges.welcomeMessage
exoplatform.challenges.message.welcomeMessageForManager==>challenges.welcomeMessageForManager
exoplatform.challenges.message.welcomeMessageForManager==>challenges.welcomeMessageForManager

File: challenges-webapp/src/main/webapp/vue-app/challenges/components/Challenges.vue [hunk 7]
```diff
@@ -1,8 +1,37 @@
...
+      color="transparent"
+      flat
+      class="pa-4">
+      <add-challenge v-if="canAddChallenge" />
+      <v-spacer />
+    </v-toolbar>
+    <template>
+      <welcome-message
+        v-if="!displayChallenges"
+        can-add-challenge="canAddChallenge" />
```
Comment (Rdidmohamedelhadi) [inline challenges-webapp/src/main/webapp/vue-app/challenges/components/Challenges.vue]:
```suggestion
        :can-add-challenge="canAddChallenge" />
```

**Review by Rdidmohamedelhadi — CHANGES_REQUESTED** [review]

File: challenges-webapp/src/main/resources/locale/portlet/Challenges_en.properties [hunk 8]
```diff
@@ -15,7 +15,7 @@
...
 #
 
-exoplatform.challenges.button.addChallenge=Add challenge
-exoplatform.challenges.message.welcomeMessage=Welcome to Challenges App!
-exoplatform.challenges.message.welcomeMessageForManager=Here you will be able to create challenges to spice things up in your workspace and boost productivity of co'workers
-exoplatform.challenges.message.welcomeMessageForUser=There is not yet challenges, stay tuned they will be added!
+challenges.button.addChallenge=Add challenge
+challenges.message.welcomeMessage=Welcome to Challenges App!
+challenges.message.welcomeMessageForManager=Here you will be able to create challenges to spice things up in your workspace and boost productivity of co'workers
+challenges.message.welcomeMessageForUser=There is not yet challenges, stay tuned they will be added!
```
Comment (Rdidmohamedelhadi) [inline challenges-webapp/src/main/resources/locale/portlet/Challenges_en.properties]: challenges.message.welcomeMessage ==>challenges.welcomeMessageForUser
challenges.message.welcomeMessageForManager ==> challenges.welcomeMessageForManager
challenges.message.welcomeMessageForUser==> challenges.welcomeMessageForUser

File: challenges-webapp/src/main/webapp/vue-app/challenges/components/WelcomeMessage.vue [hunk 9]
```diff
@@ -2,12 +2,13 @@
   <div class="welcomeMessage">
     <img
       src="/challenges/images/Challenge_app_backg_image.png"
+      alt="challenges"
```
Comment (Rdidmohamedelhadi) [inline challenges-webapp/src/main/webapp/vue-app/challenges/components/WelcomeMessage.vue]: please add description for this img and use i18n

You can see more details here : https://www.semrush.com/blog/alt-text/

**Review by Rdidmohamedelhadi — CHANGES_REQUESTED** [review]

File: challenges-webapp/src/main/webapp/vue-app/challenges/challengesServices.js [hunk 10]
```diff
@@ -0,0 +1,14 @@
+export function canAddChallenge() {
+  return fetch(`${eXo.env.portal.context}/${eXo.env.portal.rest}/challenge/api/canAddChallenge`, {
+    headers: {
+      'Content-Type': 'application/json'
```
Comment (ahamdi) [inline challenges-webapp/src/main/webapp/vue-app/challenges/challengesServices.js]: Isn't it plain/text ?

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: challenges-webapp/src/main/webapp/vue-app/challenges/challengesServices.js (hunk 10 above)
Comment (GouadriaHanen) [reply challenges-webapp/src/main/webapp/vue-app/challenges/challengesServices.js]: done!

**Review by ahamdi — APPROVED** [review]
LGTM

**Review by Rdidmohamedelhadi — APPROVED** [review]

**Pull request merged**